File: PicPay.TypeScript/AppPicPay.ts
// #region Reference

/// <reference path="../OWeb.TypeScript/AppWebBase.ts"/>
/// <reference path="../OWeb.TypeScript/design/TemaDefault.ts"/>
/// <reference path="../OWeb.TypeScript/server/ajax/dbe/SrvAjaxDbeBase.ts"/>
/// <reference path="../OWeb.TypeScript/server/ServerBase.ts"/>
/// <reference path="../OWeb.TypeScript/server/SrvHttpBase.ts"/>
/// <reference path="server/SrvAjaxDbePicPay.ts"/>
/// <reference path="server/SrvHttpPicPay.ts"/>
/// <reference path="TemaPicPay.ts"/>

// #endregion Reference

module PicPay
{
    // #region Importações

    import AppWebBase = OWeb.AppWebBase;
    import ServerBase = OWeb.ServerBase;
    import SrvAjaxDbeBase = OWeb.SrvAjaxDbeBase;
    import SrvHttpBase = OWeb.SrvHttpBase;
    import TemaDefault = OWeb.TemaDefault;

    // #endregion Importações

    // #region Enumerados
    // #endregion Enumerados

    export class AppPicPay extends AppWebBase
    {
        // #region Constantes
        // #endregion Constantes

        // #region Atributos

        protected static _i: AppPicPay;

        public static get i(): AppPicPay
        {
            if (AppPicPay._i != null)
            {
                return AppPicPay._i;
            }

            AppPicPay._i = new AppPicPay();

            return AppPicPay._i;
        }

        // #endregion Atributos

        // #region Construtores
        // #endregion Construtores

        // #region Métodos

        protected getObjTema(): TemaDefault
        {
            return new TemaPicPay();
        }

        protected inicializar(): void
        {
            super.inicializar();

            this.strNome = "PicPay";
        }

        protected inicializarArrSrv(arrSrv: Array<ServerBase>): void
        {
            arrSrv.push(new SrvAjaxDbePicPay());
            arrSrv.push(new SrvHttpPicPay());
        }

        // #endregion Métodos

        // #region Eventos

        // #endregion Eventos
    }
}

File: Principal/AppPicPay.cs
using OBase;
using OBase.Servico;
using OData;
using OWeb;
using PicPay.DataBase;
using PicPay.Server;
using System;
using System.Collections.Generic;

namespace PicPay
{
    internal class AppPicPay : AppWebBase
    {
        #region Constantes

        #endregion Constantes

        #region Atributos

        private static AppPicPay _i;

        public new static AppPicPay i
        {
            get
            {
                if (_i != null)
                {
                    return _i;
                }

                _i = new AppPicPay();

                return _i;
            }
        }

        #endregion Atributos

        #region Construtores

        private AppPicPay()
        {
        }

        #endregion Construtores

        #region Métodos

        protected override ConfigBase getCfg()
        {
            return ConfigPicPay.i;
        }

        protected override DbeBase getDbe()
        {
            return DbePicPay.i;
        }

        protected override TemaBase getObjTema()
        {
            return new TemaPicPay();
        }

        protected override void inicializarLstSrv(List<ServicoBase> lstSrv)
        {
            lstSrv.Add(new SrvAjaxDbePicPay());
            lstSrv.Add(new SrvHttpPicPay());
        }

        protected override void inicializarLstStrConsoleLogo(List<string> lstStrConsoleLogo)
        {
            base.inicializarLstStrConsoleLogo(lstStrConsoleLogo);

            lstStrConsoleLogo.Add(@"______ _     ______           ");
            lstStrConsoleLogo.Add(@"| ___ (_)    | ___ \          ");
            lstStrConsoleLogo.Add(@"| |_/ /_  ___| |_/ /_ _ _   _ ");
            lstStrConsoleLogo.Add(@"|  __/| |/ __|  __/ _` | | | |");
            lstStrConsoleLogo.Add(@"| |   | | (__| | | (_| | |_| |");
            lstStrConsoleLogo.Add(@"\_|   |_|\___\_|  \__,_|\__, |");
            lstStrConsoleLogo.Add(@"===========================/ |");
            lstStrConsoleLogo.Add(@"                        |___/ ");
        }

        [STAThread]
        private static void Main(string[] arrStrParam)
        {
            i.iniciarConsole();
        }

        #endregion Métodos

        #region Eventos

        #endregion Eventos
    }
}